refactor: Clean up and addition of version macro to clone the repo based on the release version

presatish opened the pull request:
<!-- Expected Commit Message Description (imported automatically by GitHub) -->
<!-- Must conform to [conventional commits guidelines](https://github.com/edgexfoundry/edgex-docs/blob/main/.github/Contributing.md) -->
<!-- Expected Commit message must contain Closes/Fixes #IssueNumber statement when there is a related issue -->

<!-- Add additional detailed description of need for change if no related issue -->

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-docs/blob/main/.github/Contributing.md 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Changes have been rendered and validated locally using mkdocs-material (see edgex-docs README)

lenny-intel reviewed (approved):
LGTM

ChristianDarr-personal commented on docs_src/microservices/device/supported/device-onvif-camera/General.md:
```diff
@@ -36,7 +36,7 @@ A brief video demonstration of building and using the device service:
 
 ## Examples
-To see an example utilizing the ONVIF device service, see the [camera management](https://github.com/edgexfoundry/edgex-examples/tree/{{version}}/application-services/custom/camera-management) example in the EdgeX Examples repo.
+To see an example utilizing the ONVIF device service, refer [camera management](../../../../examples/app-service-examples/camera-management/Ch-CameraManagement.md)
```
```suggestion
To see an example utilizing the ONVIF device service, refer to the [camera management example application](../../../../examples/app-service-examples/camera-management/Ch-CameraManagement.md)
```

ChristianDarr-personal commented on docs_src/microservices/device/supported/device-usb-camera/General.md:
```diff
@@ -30,7 +30,7 @@ The figure below illustrates the software flow through the architecture componen
 
 ## Examples
-To see an example utilizing the USB device service, see the [camera management](https://github.com/edgexfoundry/edgex-examples/tree/{{version}}/application-services/custom/camera-management) example in the EdgeX Examples repo.
+To see an example utilizing the USB device service, refer [camera management](../../../../examples/app-service-examples/camera-management/Ch-CameraManagement.md)
```
```suggestion
To see an example utilizing the USB device service, refer to the [camera management example application](../../../../examples/app-service-examples/camera-management/Ch-CameraManagement.md)
```

vyshali-chitikeshi reviewed (changes requested):
Validated USB both in non-secure and secure modes with default docker images as per instructions and functionality wise it works correctly. However needs to update below things for USB service
General.md
Under getting started heading, there is {.md button} that needs to be removed. Correct hyper link. Change wording to "Getting Started" from 'Get started'

2. Deployment.md
a)Rercommend to add git checkout comamnd to get edgex compose code first, very first step says cd to compose folder
b)Under Native build instructions there is no step1, needs to update step 1 properly
c) Verify service, device, and profiles section- curl commands needs to be added for both secure and non-secure modes
d)Add rtsp credentials section needs to be updated with proper alignment and/or steps

vyshali-chitikeshi reviewed (changes requested):
One more change request:
Under Mange devices section, .md button needs to be removed next to 'Learn how to use device service' hyperlink

vyshali-chitikeshi reviewed (commented):
@presatish As per our discussion, you can ignore review comments for this PR and we can address them as part of other story/PR. Otherwise, your changes in this PR looks good except need to add get version command for native builds and I think you already doing that based on our discussion

presatish commented:
@vyshali-chitikeshi As I discussed with you we are not addressing your comments in this PR as we need a separate PR for the overall docs improvement for the camera services.

ChristianDarr-personal commented on docs_src/microservices/device/supported/device-usb-camera/walkthrough/deployment.md:
````diff
@@ -9,31 +9,47 @@ Follow this guide to deploy and run the service.
-    make build
-    ```
+   
+    1. Navigate to the Edgex compose directory.
````
Shouldn't they navigate to the device-usb-camera directory ?

vyshali-chitikeshi replied:
No. This should be change to "Navigate to edgex-compose/compose-builder directory

lenny-intel reviewed (approved):
LGTM

vyshali-chitikeshi reviewed (approved):
specific changes in this PR looks good to me

presatish commented:
> LGTM, but I just have a reminder that the numbering in ordered lists is usually best to be kept as just 1s, to make maintaining the steps easier.

Sure, will keep this in mind for future work. For now I see there are too many places already without a specific ordering apart from these PR changes.

lenny-intel reviewed (approved):
LGTM
